lib/rubygems/package/tar_header.rb

```diff
@@ -243,4 +244,12 @@ def header(checksum = @checksum)
   def oct(num, len)
     format("%0#{len}o", num)
   end
+
+  eval <<-RUBY, binding, __FILE__, __LINE__ + 1
+# frozen_string_literal: true
+
+  def copy_vals(vals)
+    #{FIELDS.map {|name| "@#{name} = vals[:#{name}]" }.join("; ")}
+  end
+  RUBY
```

Comment:
This definitely needs a comment about why. This is not your grandma's self documenting code. 👵🏻

Reply:
Or, alternatively, why convert to instance variables at all? Why not store `vals` directly and iterate FIELDS one time to create attr readers that just lookup the key in the vals hash. vals is mutable and is already allocated.

Reply:
After coming back to this 3 months later, I think I'd prefer we just write out all the assignments just like in `==`. The eval feels really complex compared to simple repetition.